sources/packages/api/src/services/student-appeal/student-appeal.service.ts

```diff
@@ -128,4 +139,162 @@ export class StudentAppealService extends RecordDataModelService<StudentAppeal>
+      .innerJoin("studentAppeal.appealRequests", "appealRequest")
+      .leftJoin("appealRequest.assessedBy", "user")
+      .leftJoin("appealRequest.note", "note")
+      .where("studentAppeal.id = :appealId", { appealId })
+      .getRawAndEntities();
+
+    const [appealWithStatus] = mapFromRawAndEntities<StudentAppealWithStatus>(
+      queryResult,
+      "status",
+    );
+    return appealWithStatus;
+  }
+
+  /**
+   * Builds the conditions to define the status of a student appeal
+   * based on the statuses of its appeal requests.
+   * @returns SQL select to be used in a query to return the appeal status.
+   */
+  private buildStatusSelect(): string {
+    return `CASE
+      WHEN EXISTS(${this.studentAppealRequestsService
+        .appealsByStatusQueryObject(StudentAppealStatus.Pending)
+        .getSql()}) THEN '${StudentAppealStatus.Pending}'
+      WHEN EXISTS(${this.studentAppealRequestsService
+        .appealsByStatusQueryObject(StudentAppealStatus.Approved)
+        .getSql()}) THEN '${StudentAppealStatus.Approved}'
+      ELSE '${StudentAppealStatus.Declined}'
+    END`;
+  }
+
+  /**
+   * Update all student appeals requests at once.
+   * @param appealId appeal if to be retrieved.
+   * @param approvals all appeal requests that must be updated with
+   * an approved/declined status. All requests that belongs to the
+   * appeal must be provided.
+   * @param auditUserId user that should be considered the one that is
+   * causing the changes.
+   */
+  async approveRequests(
+    appealId: number,
+    approvals: StudentAppealRequestApproval[],
+    auditUserId: number,
+  ): Promise<StudentAppeal> {
+    const appealToUpdate = await this.repo
+      .createQueryBuilder("studentAppeal")
+      .select([
+        "studentAppeal.id",
+        "appealRequest.id",
+        "appealRequest.appealStatus",
+        "studentAssessment.id",
+        "application.id",
+        "application.data",
+      ])
+      .innerJoin("studentAppeal.appealRequests", "appealRequest")
+      .innerJoin("studentAppeal.application", "application")
+      .leftJoin("studentAppeal.studentAssessment", "studentAssessment")
+      .where("studentAppeal.id = :appealId", { appealId })
+      .getOne();
+
+    if (!appealToUpdate) {
+      throw new CustomNamedError(
+        "Not able to find the appeal.",
+        STUDENT_APPEAL_NOT_FOUND,
+      );
+    }
+
+    if (appealToUpdate.studentAssessment) {
+      throw new CustomNamedError(
+        "An assessment was already created to this student appeal.",
+        STUDENT_APPEAL_INVALID_OPERATION,
+      );
+    }
+
+    const auditUser = { id: auditUserId } as User;
+    const auditDate = new Date();
+
+    // If a students appel has, for instance, 3 requests, all must be updated at once.
+    // This counter checks if all the requests were updated as expected.
+    let updatedRecords = 0;
+    appealToUpdate.appealRequests.forEach((appealRequest) => {
```

Comment:
It looks like it is possible for appealToUpdate to have other attributes (application, studentAssessment, etc) without appealRequests. It may be necessary to check that the length of appealRequests is greater than zero before looping.

Reply:
Resolved as we talked in the call.